student.rb

```diff
@@ -1,12 +1,17 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(id, classroom, age: nil, parent_permission: true, name: 'Unknown')
+  def initialize(id, age: nil, parent_permission: true, name: 'Unknown')
     super(id, age: age, parent_permission: parent_permission, name: name)
-    @classroom = classroom
+    @classroom = nil
   end
 
   def play_hooky
     '¯\\(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom&.students&.delete(self)
+    classroom&.students&.push(self)
+  end
 end

```

Comment:
- Good job done so far, but please follow the advice of the previous reviewer to create a custom setter for the @classroom attribute. I suggest you do the following
1. Initialize the Student class with a `classroom`  instance variable, as you did before
2. implement a getter for the classroom variable
3. Add the following code like the previous reviewer suggested:
```ruby
def classroom=(classroom)
  @classroom = classroom
  classroom.students.push(self) unless classroom.students.include?(self)
end
```